vscode/microsoft-kiota/src/kiotaInterop.ts

```diff
@@ -0,0 +1,129 @@
+import * as vscode from "vscode";
+import * as cp from 'child_process';
+import * as rpc from 'vscode-jsonrpc/node';
+
+export async function connectToKiota<T>(callback:(connection: rpc.MessageConnection) => Promise<T | undefined>): Promise<T | undefined> {
+  const childProcess = cp.spawn("C:\\sources\\github\\kiota\\src\\kiota\\bin\\Debug\\net7.0\\kiota.exe", ["rpc"],{
+    cwd: vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0 ? vscode.workspace.workspaceFolders[0].uri.fsPath : undefined
+  });
+  let connection = rpc.createMessageConnection(
+    new rpc.StreamMessageReader(childProcess.stdout),
+    new rpc.StreamMessageWriter(childProcess.stdin));
+  connection.listen();
+  try {
+    return await callback(connection);
+  } catch (error) {
+    console.error(error);
+    return undefined;
+  } finally {
+    connection.dispose();
+    childProcess.kill();
+  }
+}
+
+export interface KiotaLogEntry {
+  level: number;
+  message: string;
+}
+
+export interface KiotaOpenApiNode {
+    segment: string,
+    path: string,
+    children: KiotaOpenApiNode[],
+    selected: boolean,
+}
+
+export interface KiotaShowConfiguration {
+    includeFilters: string[];
+    excludeFilters: string[];
+    descriptionPath: string;
+}
+
+interface KiotaLoggedResult {
+    logs: KiotaLogEntry[];
+}
+export interface KiotaShowResult extends KiotaLoggedResult {
+    rootNode?: KiotaOpenApiNode;
+}
+
+export interface KiotaSearchResult extends KiotaLoggedResult {
+    results: Record<string, KiotaSearchResultItem>;
+}
+export interface KiotaSearchResultItem {
+  Title: string;
+  Description: string;
+  ServiceUrl?: string;
+  DescriptionUrl?: string;
+  VersionLabels?: string[];
+}
+
+export enum KiotaGenerationLanguage {
```

Comment:
I'm wondering if we could save some maintenance work here by using a string enum. 
https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

Forward lookup would give us the string directly eliminating the work to update `generationLanguageToString` function and the `parseGenerationLanguage` would simply be lookup of the object entries with something like
```js
function parseGenerationLanguage(enumString: string): Languages {
    const match = Object.entries(Languages)
                        .find(([_, value]) => value === enumString);
    if (match) {
        const [_,value] = match;
        return value
    }
    throw new Error("unknown language");
}
```

Reply:
but if we were to do that, the JSON RPC communication would be broken I believe (kiota would send ints, typescript deserialization would expect strings)

Reply:
Would this be the case for string enums as well? I figure their values would strings not ints

Reply:
Sorry, I understand now. Resolving this now. I kept understanding the int requirement coming from the Typescript end. 
Resolving